File: examxy.Application/Features/Realtime/RealtimeEventTypes.cs
namespace examxy.Application.Features.Realtime
{
    public static class RealtimeEventTypes
    {
        public static class Notification
        {
            public const string Created = "notification.created";
            public const string Read = "notification.read";
        }

        public static class Post
        {
            public const string Created = "post.created";
            public const string Updated = "post.updated";
        }

        public static class Comment
        {
            public const string Created = "comment.created";
            public const string Updated = "comment.updated";
            public const string Hidden = "comment.hidden";
        }

        public static class Reaction
        {
            public const string PostUpdated = "reaction.post.updated";
            public const string CommentUpdated = "reaction.comment.updated";
        }
    }
}

File: examxy.client/src/features/realtime/lib/realtime-event-types.ts
export const realtimeEventTypes = {
  notification: {
    created: 'notification.created',
    read: 'notification.read',
  },
  post: {
    created: 'post.created',
    updated: 'post.updated',
  },
  comment: {
    created: 'comment.created',
    updated: 'comment.updated',
    hidden: 'comment.hidden',
  },
  reaction: {
    postUpdated: 'reaction.post.updated',
    commentUpdated: 'reaction.comment.updated',
  },
} as const

export const realtimeScopeTypes = {
  user: 'user',
  class: 'class',
} as const

export const realtimeHubMethods = {
  receiveRealtimeEvent: 'ReceiveRealtimeEvent',
  subscribeClass: 'SubscribeClass',
  unsubscribeClass: 'UnsubscribeClass',
} as const
